File: backend/src/main/java/com/example/bookhub/dto/BookDto.java
package com.example.bookhub.dto;

import com.example.bookhub.enums.Genre;
import com.example.bookhub.enums.Language;
import lombok.Data;

@Data
public class BookDto {

    private Long id;

    private String title;

    private String author;

    private String publisher;

    private String isbn;

    private int publicationYear;

    private Genre genre;

    private int pageCount;

    private Language language;

    private double averageRating;

    private int numberOfRatings;

    private String description;

    private String imagePath;
}

File: frontend/src/app/core/services/admin/models/book.dto.ts
import { EbookFormat } from '../../../../shared/enums/ebook-format';
import { Genre } from '../../../../shared/enums/genre';
import { Language } from '../../../../shared/enums/language';

export interface BookDto {
  readonly id: number;
  readonly title: string;
  readonly author: string;
  readonly publisher: string;
  readonly isbn: string;
  readonly publicationYear: number;
  readonly genre: Genre;
  readonly pageCount: number;
  readonly language: Language;
  readonly description: string;
  readonly imagePath: string;
  readonly isEbook: boolean;
  readonly ebookFormat: EbookFormat | null;
  readonly ebookFileSize: number | null;
  readonly ebookLink: string;
}
